File: ui/src/types/data-request.ts
export interface DataRequest {
  id: number
  first_name: string
  last_name: string
  status: number
  created_on: string
  created_by: string
  request_source_id: string
}

export const StatusLabels: Record<number, string> = {
  1: 'Created',
  2: 'Processing',
  3: 'Needs Review',
  99: 'Complete',
}

File: backend/core/data_request/data_request.py
from dataclasses import dataclass
from datetime import date, datetime
from enum import IntEnum


class Status(IntEnum):
    """Status values for data requests."""

    CREATED = 1
    PROCESSING = 2
    NEEDS_REVIEW = 3
    COMPLETE = 99


@dataclass
class DataRequest:
    """Data transfer object for a data request."""

    id: int
    person_id: int
    first_name: str
    last_name: str
    date_of_birth: date
    status: Status
    created_on: datetime
    created_by: str
    request_source_id: str
